src/sentry/snuba/metrics/datasource.py

```diff
@@ -671,6 +673,16 @@ def get_series(
     include_meta: bool = False,
 ) -> dict:
     """Get time series for the given query"""
+
+    if metrics_query.interval is not None:
+        interval = metrics_query.interval
+    else:
+        interval = metrics_query.granularity.granularity
+
+    start, end, _num_intervals = to_intervals(metrics_query.start, metrics_query.end, interval)
+
+    metrics_query = replace(metrics_query, start=start, end=end)
+
```

Comment:
Why are we changing the time bounds of the query, I think I am missing something? I thought we didn't want to do that, especially since the metrics layer did not adjust these bounds previously.

E.g. if you make the query from 09:30:00 to 10:30:00 with granularity 60 and interval 3600 then the intervals will be [09:00-10:00, 10:00-11:00] that will cause the new time bounds to be 9:00 to 11:00. Is this what happens?

Reply:
Yes, your remarks are correct. 
It is a compromise that we came up with, from the conclusion of the "fix intervals document":

> **The desired behaviour is to include all intervals that have a non null intersection with the passed date range.** 
> 
> **We agreed that not only the intervals will be changed but also the querying range will include the whole adjusted date range from the beginning of the first interval to the end of the last**, so at the entry into the API, i.e. `get_series` the start, end parameters will be aligned. **Any function inside the metric layer is free to assert that start & end are properly aligned to interval and that interval is a multiple of granularity.**